### ISSUE TITLE
Error thrown for task marked with noqa suppression

### ISSUE THREAD

**Issue by nre-ableton** [ansible-community/ansible-lint#1367]
##### Summary

In one of our playbooks, we have the following task:

```yaml
- name: Display list of changes
  delegate_to: localhost
  command: "git log --no-merges --oneline {{ host_revision.0 }}..HEAD"  # noqa command-instead-of-module
  register: git_log_cmd
  changed_when: false
```

ansible-lint 5.0.0 prints `diff.yml:32: command-instead-of-module git used in place of git module`. I have not yet tried on version 5.0.1, since it seems the release hasn't yet found its way to pypi, however I also verified the behavior on commit 9dbccb8f19addeda53ac7eb6d52ac000b9698032 in the `master` branch. I also tried moving the suppression to the same line as the task name, but that didn't work either.

Strangely, we have a handful of other similar tasks in other places which also have the `noqa command-instead-of-module` suppression and are correctly ignored. The only difference I can spot is that this task is in a top-level playbook and not part of a role or other task file.

##### Issue Type

- Bug Report

##### Ansible and Ansible Lint details

```console
$ ansible --version
ansible 2.10.5
  config file = /home/nre/<redacted>/ansible.cfg
  configured module search path = ['/home/nre/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/nre/<redacted>/lib/python3.8/site-packages/ansible
  executable location = /home/nre/<redacted>/bin/ansible
  python version = 3.8.5 (default, Jul 28 2020, 12:59:40) [GCC 9.3.0]
$ ansible-lint --version
ansible-lint 5.0.0
```

- ansible installation method: pip
- ansible-lint installation method: pip

**Comment by nre-ableton** [comment]
One thought -- I noticed that our top-level playbooks are being parsed as type `yaml` (see https://github.com/ansible-community/ansible-lint/blob/53c8aeefaab95071467576ac732a42f75f70bce8/src/ansiblelint/config.py#L13). Would this have anything to do with it?

Also, why does ansible-lint not consider top-level YAML files as playbooks? I have not seen any recommendations by Ansible for placing playbooks in a `playbooks` folder (see https://docs.ansible.com/ansible/latest/user_guide/sample_setup.html).

**Comment by ssbarnea** [comment]
Your remark is right and it should worry you: parsed as YAML it means only yamllint will be run on them, nothing ansible-specific.

If you dig among issue comments, discussions and irc history you will find lots of things around that subject. Mainly Ansible itself does not care about anything that is YAML but is not really an Ansible managed file. 

Keep in mind that Ansible examples are there to exemplify specific use aspects and they do miss to follow best practices in order to keep the examples as small as possible.

This means that if you have docker-compose.yml at your repository root, it could as well be an Ansible playbook unless you start putting playbooks in their own folder.

There reality is that, in real life, most people will need to host both playbooks and other random YAML files inside the same repository. Without following a standard structure confusions will appear and it will be impossible for a tool to guess the file type. 

As you probably already seen, **collections layout** already covered for this and indicates clearly where the playbooks are to be put.

Collection layout is not only for collections that you may want to publish or not, is for keeping track of any ansible code you wrote.

**Comment by nre-ableton** [comment]
Thanks for the explanation; I'll have a look at converting our project to the collection layout.
